pymc/distributions/distribution.py

```diff
@@ -687,7 +756,7 @@ def custom_dist_logcdf(op, value, size, *params, **kwargs):
 
         @_moment.register(rv_type)
         def custom_dist_get_moment(op, rv, size, *params):
-            return moment(rv, size, *params[: len(params)])
+            return moment(rv, size, *filter_RNGs(params))
```

Comment:
Only register this when `moment is not None`. And the default case you can register in the base class `CustomSymbolicDistRV` outside